### PR TITLE
feat(monitoring): connect monitoring contract to multisig prover

### PR DESCRIPTION
**Pull request by maancham**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: integration-tests/tests/unbond_worker.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+use cosmwasm_std::{Addr, StdError};
+use integration_tests::contract::Contract;
+use std::collections::HashSet;
+
+pub mod test_utils;
+
+#[test]
+fn worker_should_be_in_collective_active_worker_set_of_all_supported_chains() {
+    let (protocol, ethereum, polygon, initial_workers, _) = test_utils::setup_test_case();
```
Comment (cgorenflo) [inline integration-tests/tests/unbond_worker.rs:9]: not in scope for this PR, but something to think about: `setup_test_case` hides too much detail. I can't really see what this test does without digging three layers deep. We should remodel that, maybe with a builder pattern like so:
```
let setup = set_up_test_case()
    .with_workers(...)
    .with_chains(...)

```

File: integration-tests/tests/unbond_worker.rs
Comment (cgorenflo) [inline integration-tests/tests/unbond_worker.rs:1]: all the tests have the same structure:
- a set of chains
- a worker address
- an expected outcome

you could parameterize this instead of writing a separate test for each (see https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/gateway/tests/contract.rs)

Additionally, because this is a critical query for unbonding, we need to catch all the edge cases:
- chain is supported, but worker is not part of the active set (but supports other chains)
- call only with some of the known chains
- worker was in a previous worker set for chain, but not the currently active one

This can become quite difficult to set up with integration tests (because it involved a lot of verification, rotation etc), so it might we easier to set it up as unit tests and mocks for the queries to other contracts. Definitely have some integration tests to make sure the contract endpoints work together though

**Review by cgorenflo — CHANGES_REQUESTED** [review]

File: contracts/monitoring/src/query.rs [hunk 2]
```diff
@@ -16,3 +18,30 @@ pub fn provers(deps: Deps, chain_name: ChainName) -> Result<Vec<Addr>, ContractE
...
 }
+
+pub fn check_worker_is_active(
+    deps: Deps,
+    chain_names: HashSet<ChainName>,
+    worker: Addr,
+) -> StdResult<bool> {
+    for chain_name in chain_names {
+        if let Ok(prover_addresses) = PROVERS_PER_CHAIN.load(deps.storage, chain_name) {
+            for prover_address in prover_addresses {
```
[the remark this answers is not in the document]
Comment (maancham) [reply contracts/monitoring/src/query.rs:29]: ok thanks for double checking!

File: integration-tests/tests/unbond_worker.rs (hunk 1 above)
Comment (maancham) [reply integration-tests/tests/unbond_worker.rs:9]: Ok sure. I will add a ticket for it to the backlog to later come back to this.
